drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,256 @@
+
+Expiry Height
+-------------
+
+The expiry height field specifies the last block height at which a transaction must be mined into a block, after which the transaction is deemed to have expired and should be removed from the mempool.  A block is invalid if it contains an expired transaction.  More information can be found in the Transaction Expiry ZIP [#zip-???]_
+
+Transaction Validation
+======================
+
+A client which supports the consensus rules identified by the format branch id should accept the transaction for further validation.  If the value of the branch id is zero, this indicates the absence of any network upgrade and that Sprout consensus rules apply.
+
+Overwinter parsers should reject transactions from further processing and validation if the:
+
+- version number is unknown
+- overwinter flag is not set
+- format branch id is not set
+
+Overwinter clients should reject transactions for violating consensus rules if:
+
+- version number is invalid according to current set of active consensus rules
+- format branch id is different from the branch id corresponding to the current set of active consensus rules
+- mis-matched format branch id causes signatures to fail verification under the new Overwinter signature hashing scheme which includes a branch id
+
+Implementation
+==============
+
+Transaction Version
+-------------------
+
+Transaction version remains a positive value.
+
+Code can continue to check the transaction version using comparison operators::
+
+    if (tx.nVersion >= 2) {
+        for (int js = 0; js < joinsplits; js++) {
+
+Tests can continue to set the version to zero as an error condition::
+
+    mtx.nVersion = 0
+    
+Overwinter Validation
+---------------------
+
+To test if the format of an Overwinter transaction is valid or not::
+
+    if (tx.nVersion == 3 && tx.fOverwintered) {
+        // Valid v3 format transaction
+        // FIXME: If tx.nFormatBranchId is 0, does that make it invalid?
+    }
+
+To test if the format of an Overwinter transaction is intended for the client's chain::
+
+    if (tx.nVersion == 3 &&
+        tx.fOverwintered &&
+        tx.nFormatBranchId == NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId) {
+        // Valid for the client's chain
+    }
+
+Deployment
+==========
+
+This proposal will be deployed with the Overwinter network upgrade.
+
+Testnet is set to activate Overwinter at block XXX.
+
+- This means that starting from block XXX of testnet, new Overwinter consensus rules take effect and transactions must be using v3 to be accepted as valid.
+
+Mainnet is set to activate Overwinter at block XXX.
+
+- This means that starting from block XXX of mainnet, new Overwinter consensus rules take effect and transactions must be using v3 to be accepted as valid.
```

Comment:
Hmm, we haven't actually decided where we will specify the Overwinter activation heights. Should we pick one of the Overwinter ZIPs (like here) for it, or have another trivial (meta-)ZIP for Overwinter itself? @daira?

Reply:
Let's put them in ZIP 201 (just because that's the first ZIP specific to Overwinter in intended reading order), and have ZIPs {143, 202, 203} refer back to ZIP 201.

Reply:
Actually there's already a (currently stub) Deployment section in ZIP 201 which is what this should reference.